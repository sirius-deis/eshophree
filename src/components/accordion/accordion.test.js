import { render, screen, fireEvent } from '@testing-library/react';
import Accordion from './accordion';

describe('Accordion component', () => {
  const title = 'Accordion Title';
  const content = 'Accordion Content';
  it('should match snapshot with kind round', () => {
    const { container } = render(<Accordion content={content} title={title}/>);
    expect(container).toMatchSnapshot();
  });
  it('should render the title correctly', () => {
    render(<Accordion title={title} content={content} />);
    expect(screen.getByText(title)).toBeInTheDocument();
  });

  it('should not display content initially', () => {
    render(<Accordion title={title} content={content} />);
    expect(screen.queryByText(content)).not.toBeInTheDocument();
  });

  it('should toggle content visibility when button is clicked', () => {
    render(<Accordion title={title} content={content} />);
    const button = screen.getByRole('button', { name: /accordion title/i });

    fireEvent.click(button);
    expect(screen.getByText(content)).toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.queryByText(content)).not.toBeInTheDocument();
  });

  it('should display the correct icon based on the state', () => {
    render(<Accordion title={title} content={content} />);
    const button = screen.getByRole('button', { name: /accordion title/i });

    expect(screen.getByText('+')).toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByText('âˆ’')).toBeInTheDocument();
  });
});