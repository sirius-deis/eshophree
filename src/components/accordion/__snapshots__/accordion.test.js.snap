// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component should match snapshot with kind round 1`] = `
<div>
  <div
    class="sc-bgqQcB jIiUBw"
  >
    <div
      class="sc-gTRrQi hnLgNS"
    >
      <button
        class="sc-ewnqHT gA-dPuy"
      >
        <span>
          Accordion Title
        </span>
         
        +
      </button>
    </div>
  </div>
</div>
`;
